### PR TITLE
Add the ability to create new ruleset and new rule

### LINKED ISSUE #35: CLI - create new ruleset
Some type of command to initialize / create a new ruleset project

### PR DESCRIPTION
**Pull request by sewe2000**
Resolves #35

### CODE REVIEW

File: src/checker/ruleset_location.go [hunk 1]
```diff
@@ -70,34 +70,34 @@ func (r *RuleSetLocation) new(identifier string) {
 	r.simpleUrl = simpleUrl
 }
 
-func (r *RuleSetLocation) createDir() error {
-	println(r.getRuleSetPath())
-	err := os.MkdirAll(r.getRuleSetPath(), 0700)
+func (r *RuleSetLocation) CreateDir() error {
+	println(r.GetRuleSetPath())
```
Comment (maxidragon) [inline src/checker/ruleset_location.go:74]:
```suggestion
```

**Review by cooligus — COMMENTED** [review]
I suggest to change names of these commands:
`nr` or `new rule` - to create rule
`ns` or `new set` - to create set

**Comment by sewe2000** [comment]
> I suggest changing names of these commands: `nr` or `new rule` - to create rule `ns` or `new set` - to create set

@BaderBC What do you think about it?

**Comment by BaderBC** [comment]
> > I suggest changing names of these commands: `nr` or `new rule` - to create rule `ns` or `new set` - to create set
> 
> @BaderBC What do you think about it?

Even though It sounds good I'm not sure whether it is necessary, especially that `new rule` isn't the command which would be often used.

**Comment by sewe2000** [comment]
> > > I suggest changing names of these commands: `nr` or `new rule` - to create rule `ns` or `new set` - to create set
> > 
> > 
> > @BaderBC What do you think about it?
> 
> Even though It sounds good I'm not sure whether it is necessary, especially that `new rule` isn't the command which would be often used.

The thing is I've implemented the process of creating a new rule, but the command is ```generate {rule_name}````. @cooligus suggested to change it to ```new rule {rule_name}```

**Comment by BaderBC** [comment]
> > > > I suggest changing names of these commands: `nr` or `new rule` - to create rule `ns` or `new set` - to create set
> > > 
> > > 
> > > @BaderBC What do you think about it?
> > 
> > Even though It sounds good I'm not sure whether it is necessary, especially that `new rule` isn't the command which would be often used.
> 
> The thing is I've implemented the process of creating a new rule, but the command is ```generate {rule_name}````. @cooligus suggested to change it to ```new rule {rule_name}```

Okay,
Now I understand. Personally i prefer to do it the same way Nest.js cli did it - `generate`

**Review by cooligus — APPROVED** [review]

File: src/cmd/generate.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+	}
+	handleError(error)
+
+	config := ConfigFileLayout{}
+	error = yaml.Unmarshal(configFileContents, &config)
+	handleError(error)
+
+	rulesetLocation := checker.RuleSetLocation{}
+	rulesetLocation.New(config.Identifier)
+	if !rulesetLocation.IsRuleSetDownloaded() {
```
Comment (BaderBC) [inline src/cmd/generate.go:33]: I don't understand this line
When someone is downloading a new ruleset it does makes sens that it needs to be putted into specific directory.
But if someone just wants to create a new rule in ruleset it should look like a normal project, meaning that it should be allowed to create ruleset wherever you want to.

File: src/cmd/generate.go [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+		printErrorAndExit(errors.New("Your ruleset is not in the standard ruleset path!"))
+	}
+
+	ruleName := strings.ReplaceAll(args[0], " ", "")
+	ruleFile, error := os.Create("rules/" + ruleName + ".lua")
```
Comment (BaderBC) [inline src/cmd/generate.go:38]: I think it should be possible to create new rule in subdirectory e.g. create rule in `rules/example/dir/rule.lua`

File: src/cmd/new.go [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+	"gopkg.in/yaml.v3"
+)
+
+var newRulesetCommand = &cobra.Command{
+	Use:   "new {ruleset_name}",
+	Short: "Create new ruleset",
+	Args:  cobra.ExactArgs(1),
+	Run: func(cmd *cobra.Command, args []string) {
+		rulesetName := strings.ReplaceAll(args[0], " ", "")
+		gitCommand := exec.Command("git", "config", "--global", "--list")
```
Comment (BaderBC) [inline src/cmd/new.go:22]: I think it is better idea to use library for git (especially that checker module already imports library for git)

File: src/cmd/root.go [hunk 5]
```diff
@@ -24,7 +45,13 @@ func Execute() {
...
 	rootCmd.AddCommand(checkFileCmd)
+	rootCmd.AddCommand(newRulesetCommand)
+	rootCmd.AddCommand(generateRuleCommand)
+	rootCmd.AddCommand(generateShortCommand)
 
 	checkFileCmd.Flags().Bool("gui-child-mode", false, "Tells app that it is executed by gui")
 	checkCmd.Flags().Bool("gui-child-mode", false, "Tells app that it is executed by gui")
+	newRulesetCommand.Flags().Bool("gui-child-mode", false, "Tells app that it is executed by gui")
+	generateRuleCommand.Flags().Bool("gui-child-mode", false, "Tells app that it is executed by gui")
+	generateShortCommand.Flags().Bool("gui-child-mode", false, "Tells app that it is executed by gui")
```
Comment (BaderBC) [inline src/cmd/root.go:56]: I don't think it is needed for your command, because I created this flag only in order to redirect all api.info.* to dbus

**Review by BaderBC — CHANGES_REQUESTED** [review]

**Review by BaderBC — CHANGES_REQUESTED** [review]
I have only one question:  Have you ever tested it?
image

**Review by BaderBC — CHANGES_REQUESTED** [review]
`spito new test-ruleset` generated this `spito-rules.yml` file:
```
repo_url: https://github.com/Bartłomiej Strama/test-ruleset
git_prefix: github.com
identifier: Bartłomiej Strama/test-ruleset
rules:
    example-rule: ./rules/dir2/subdir/example-rule.lua
```

Because of repo_url value, which it automaticly created, I think after executing `spito new` we should ask few questions like: what is your repo url (simillary to what happens when you execute `npm init`)
Don't forget to add option `-y` which will choose default `spito-rules.yml` values

**Review by BaderBC — APPROVED** [review]

**Pull request merged**